app/src/main/java/com/lightricks/feedexercise/network/FeedApiService.kt

```diff
@@ -1,5 +1,39 @@
 package com.lightricks.feedexercise.network
 
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import io.reactivex.Single
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.GET
+
+
 /**
  * todo: add the FeedApiService interface and the Retrofit and Moshi code here
  */
+
+
+interface FeedApiService {
+    @GET("Android/demo/feed.json")
+    fun getFeed():  Single<FeedItemDto.FeedItemDtoList>
+
+    companion object NetworkAccess {
```

Comment:
I would remove the `NetworkAccess` name you gave to the companion object. I think accessing the instance from the `FeedApiService` directly is better, ```FeedApiService.feedApiService``` (by the way we can rename `feedApiService` by `instance`) instead of `FeedApiService.NetworkAccess.feedApiService`